libglusterfs/src/inode.c

```diff
@@ -735,7 +735,12 @@ inode_forget_atomic(inode_t *inode, uint64_t nlookup)
         GF_ATOMIC_INIT(inode->nlookup, 0);
     } else {
         inode_lookup = GF_ATOMIC_FETCH_SUB(inode->nlookup, nlookup);
-        GF_ASSERT(inode_lookup >= nlookup);
+        //GF_ASSERT(inode_lookup >= nlookup);
+	if (inode_lookup < nlookup) {
+	    GF_ATOMIC_FETCH_ADD(inode->nlookup, nlookup - inode_lookup);
+	    gf_msg_callingfn(THIS->name, GF_LOG_WARNING, 0, LG_MSG_ASSERTION_FAILED,
+			     "GF_ASSERT(inode_lookup >= nlookup) reset inode->nlookup to 0");
```

Comment:
Probably it would be better to add a reference to a potential kernel issue to point the user in the right direction.

Reply:
ok